services/py-api/src/database/repository/teams_repository.py

```diff
@@ -29,10 +30,21 @@ async def create(
         **kwargs: Dict[str, Any]
     ) -> Result[Team, DuplicateTeamNameError | Exception]:
 
+        team = Team(name=input_data.team_name)
+
+        async def db_operation() -> None:
+            await self._collection.insert_one(document=team.dump_as_mongo_db_document(), session=session)
```

Comment:
Wouldn't it have made sense to actually create to have an abstraction on top of collection which accepts the actual pydantic object (or whatever you're using).

Reply:
We are using custom data classes due to serialization problems with the ObejctID. You can check them [here](https://github.com/AUBGTheHUB/monolith/tree/main/services/py-api/src/database/model). TBH, I never thought about such abstraction, I thought the repository layer was good enough, and creating the types inline was just fine. We could think about your idea, but it will require a lot of changes.

Reply:
I'm too lazy to explain why this is so important but the TL;DR - Better DevEx, less prone to errors because dump happens in the base concrete class of `MongoCollection` which is extended by all of the collections -> you get to only have to deal with your custom model objects and not manually dump them. Do not consider this a mandatory change by any means. I'm just pointing it out.